### PR TITLE
chore(ci): split e2e tests into configs for perf vs non-perf

### PR DESCRIPTION
**Pull request by juliemr**
This should make it easier to run perf tests versus example e2e tests,
and help stabilize the travis build because perf tests always
run with a clean browser.

### CODE REVIEW

**Comment by juliemr** [comment]
New commit message:

chore(ci): reorganize e2e/perf test running …

Now, running protractor configs by default only runs e2e tests. If
the --benchmark flag is added, it runs only the perf tests, and always
restarts the browser in between tests. If the --dryrun test is added,
the perf tests are run only once.

This should make it easier to run perf tests versus example e2e tests,
and help stabilize the travis build because perf tests always
run with a clean browser.

File: protractor-shared.js [hunk 1]
```diff
@@ -225,7 +229,7 @@ exports.createBenchpressRunner = function(options) {
     benchpress.JsonFileReporter.BINDINGS,
     benchpress.bind(benchpress.JsonFileReporter.PATH).toValue(resultsFolder)
   ];
-  if (argv['benchmark']) {
+  if (argv['benchmark'] && !argv['dryrun']) {
```
Comment (tbosch) [inline protractor-shared.js]: You don't need to check for the `benchmark` here, only for `dryrun`.

File: protractor-shared.js [hunk 2]
```diff
@@ -145,7 +149,7 @@ var config = exports.config = {
     }
   },
```
Comment (tbosch) [inline protractor-shared.js]: I think you are missing the `restartBrowserBetweenTests` flag, no?

Comment (juliemr) [inline protractor-shared.js]: Changed as discussed offline - now using this flag instead of manually doing it.

File: protractor-shared.js (hunk 1 above)
Comment (juliemr) [inline protractor-shared.js]: done.

**Comment by juliemr** [comment]
Note: I had to revert to using the `onPrepare` hacks to open up new browsers due to https://github.com/angular/protractor/issues/1983

I've added a todo to investigate that. For the moment, this is all passing locally. Just waiting for Travis to go green.

**Comment by juliemr** [comment]
Green! Merged as 3afb744e7765cfe99fbb14974827cbc5617cc84b

**Pull request closed without merging**